More 2.0 startup fixes

SophieFitz opened the pull request:
This is all the startup fixes now! Angel's now loads into a new game.

Quite a few small fixes. The biggest one is for `icon_size`.
See the commit descriptions for full info :)

Still needing attention: entity `sea-pump-placeable`.

SophieFitz committed:
icon_size fixes

- The icon_size for all sprites now defaults to 64.
- That means that a lot of (mainly recipe) 'icons' definitions need their
  icon_size manually set to 32.
- It also needs setting per icon (if the 'icons' table is defined).
  Setting icon_size outside of 'icons' no longer works.

- I'm collating these fixes under one commit because new
  higher resolution icons may be created in the future.
  This commit can then be undone to accommodate.

SophieFitz committed:
More startup fixes

Data stage:
- Fix naming issue in bio-tile.lua
- Tech 'advanced-electronics' is now named after the item it unlocks,
  namely 'advanced-circuit'.
  Same for 'advanced-electronics-2' --> 'processing-unit'
  And 'optics' --> 'lamp'

- Load alien artifacts only if bobsenemies is active.
- Fix oil refinery collision mask
- Remove flow_direction from valves (2.0 broke this)
- 'rock-huge.png' renamed to 'huge-rock.png'
- Commented out resource removal of base game ores
- Workaround fixes for trigger techs (angel's smelting data-final-fixes.lua)

Control stage:
- Rename sprite 'utility.close_white'
- Commented out Gui element with style 'draggable_space_with_no_left_margin'

KiwiHawk commented on angelsbioprocessing/data.lua:
```diff
@@ -12,6 +12,8 @@ angelsmods.trigger.lab_ignore_token["lab-module"] = true
 --ARTIFACTS
 angelsmods.trigger.artifacts = angelsmods.trigger.artifacts or {}
 if bobmods and bobmods.enemies and data.raw.item["small-alien-artifact-blue"] then
+  require("prototypes.technology.bio-processing-artifacts") -- Only load artifacts if bobs is enabled (causes startup halt without this)
```
Conditional require statements should be avoided. They are a common cause of desyncs. Better to have the condition in the required file. Make it immediately exit if the condition is not true.

KiwiHawk commented on angelsbioprocessing/prototypes/items/bio-refugium-biter.lua:
```diff
@@ -36,15 +37,16 @@ data:extend({
       {
         icon = "__angelsbioprocessinggraphics__/graphics/icons/biter-crown.png",
+        icon_size = 32,
         icon_size = 15,
```
icon_size defined twice?

KiwiHawk commented on angelsbioprocessing/prototypes/items/bio-refugium-biter.lua:
```diff
@@ -64,15 +66,16 @@ data:extend({
       {
         icon = "__angelsbioprocessinggraphics__/graphics/icons/biter-crown.png",
+        icon_size = 32,
         icon_size = 15,
```
icon_size defined twice?

KiwiHawk commented on angelsbioprocessing/prototypes/items/bio-refugium-biter.lua:
```diff
@@ -8,15 +8,16 @@ data:extend({
       {
         icon = "__angelsbioprocessinggraphics__/graphics/icons/biter-crown.png",
+        icon_size = 32,
         icon_size = 15,
```
icon_size defined twice?

KiwiHawk commented on angelsrefining/prototypes/generation/angels-override.lua:
```diff
@@ -1,9 +1,11 @@
-  angelsmods.functions.remove_resource("iron-ore")
-  angelsmods.functions.remove_resource("copper-ore")
-  angelsmods.functions.remove_resource("uranium-ore")
+  -- These throw this error:
```
Please add TODO to the comment here to make it easier to locate

KiwiHawk commented on angelsrefining/src/welcome_dialog.lua:
```diff
@@ -150,15 +150,15 @@ function welcome_dialog:create_welcome_dialog(player_index)
-  })
-  footerSpace.style.horizontally_stretchable = true
-  footerSpace.style.height = 32
+--   local footerSpace = welcomeFrameFooter.add({
```
This is just temporarily commented out I assume? If so, please add TODO comment

KiwiHawk commented on angelssmelting/data-final-fixes.lua:
```diff
@@ -2,3 +2,13 @@ require("prototypes.smelting-override-global")
 -- EXECUTE OVERRIDES
 angelsmods.functions.OV.execute()
+
+--TEMPORARY OVERRIDE FOR TECHS WITH TRIGGERS (CAUSES A STARTUP HALT)
```
Please add TODO comment

SophieFitz committed:
Resolve PR review requests

- Add TODOs
- Move conditional for alien artifacts to within file
  (instead of within conditional require statement)
- Remove extra (and incorrect) icon_size definitions

SophieFitz commented:
Should be good now
